packages/server/src/tests/utilities/api/base.ts

```diff
@@ -1,17 +1,164 @@
 import TestConfiguration from "../TestConfiguration"
 import { SuperTest, Test } from "supertest"
+import { ReadStream } from "fs"
 
-export interface TestAPIOpts {
-  headers?: any
+type Headers = Record<string, string | string[] | undefined>
+
+export interface AttachedFile {
+  name: string
+  file: Buffer | ReadStream | string
+}
+
+function isAttachedFile(file: any): file is AttachedFile {
+  if (file === undefined) {
+    return false
+  }
+  const attachedFile = file as AttachedFile
+  return (
+    Object.hasOwnProperty.call(attachedFile, "file") &&
+    Object.hasOwnProperty.call(attachedFile, "name")
+  )
+}
+
+export interface Expectations {
```

Comment:
When this is done - i'd probably move these into `@budibase/types` into some testing types - we should probably reuse these ideas in say the worker tests as well.